### ISSUE TITLE
Possible bug when computing the log-det of Jacobian for affine coupling

### ISSUE THREAD

**Issue by wubowen416** [simonalexanderson/StyleGestures#33]
Hi, thank you for providing the code, it really helps a lot.

The problem(possible) lies here. 
https://github.com/simonalexanderson/StyleGestures/blob/a7599355a7b6578dba96e5ec76cc7cfeedd272e8/glow/models.py#L110

Since the coupling layer is applied to each time step, i.e., on input tensor of size `(N, C, T)`, the Jacobian of this operation should be a partitioned matrix, which is a diagonal matrix of `diag(s)`, where s is the scale matrix. 

As a result, the determinant of the Jacobian shoule be `det(s)^T`, the log-det should be `T*log(det(s))`.

Thus, the first part of the code above `thops.sum(torch.log(scale), dim=[1, 2])` should be timed with T.

The possible effect might be that the weight matrix for focasting hidden state of LSTM wouldn't well trained, since the amplitude of the loss on all timesteps was shrinked unexpectly.

**Comment by wubowen416** [comment]
Sorry it was a misunderstading.

**Comment by simonalexanderson** [comment]
Hey @wubowen416 ,
Thanks for reaching out. I am very happy you are scrutinizing the code, it really helps getting better quality research. Don't hesitate to get back if you have any questions or concerns. Cheers, Simon
